drivers/rtc/rtc_atmel_sam.c

```diff
@@ -0,0 +1,711 @@
+/*
+ * RTC driver for the ATMEL SAM family
+ *
+ * Copyright 2022 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT atmel_sam_rtc
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/rtc.h>
+
+#include <time.h>
+#include <string.h>
+#include <soc.h>
+
+#define RTC_ATMEL_SAM_TO_BCD(value) \
+	(((((uint32_t)(value)) / 10) << 4) + ((value) % 10))
+
+#define RTC_ATMEL_SAM_FROM_BCD(value) \
+	((((value) >> 4) * 10) + ((value) & BIT_MASK(4)))
+
+#define RTC_ATMEL_SAM_REG_GET_FIELD(value, field) \
+	((RTC_##field##_Msk & value) >> RTC_##field##_Pos)
+
+#define RTC_ATMEL_SAM_REG_GET_FIELD_BCD(value, field) \
+	RTC_ATMEL_SAM_FROM_BCD(RTC_ATMEL_SAM_REG_GET_FIELD(value, field))
+
+#if defined(REG_RTC_WPMR)
+#define RTC_ATMEL_SAM_REG_RTC_WPMR REG_RTC_WPMR
+#elif defined(REG_SYSC_WPMR)
+#define RTC_ATMEL_SAM_REG_RTC_WPMR REG_SYSC_WPMR
+#else
+#define RTC_ATMEL_SAM_REG_RTC_WPMR (*(((uint32_t *)SUPC) + 53))
+#endif
+
+#define RTC_ATMEL_SAM_WPMR_DISABLE 0x52544300
+#define RTC_ATMEL_SAM_WPMR_ENABLE 0x52544300
+
+typedef void (*rtc_atmel_sam_irq_init_fn_ptr)(void);
+
+struct rtc_atmel_sam_config {
+	Rtc *regs;
+	uint16_t irq_num;
+	rtc_atmel_sam_irq_init_fn_ptr irq_init_fn_ptr;
+};
+
+struct rtc_atmel_sam_data {
+	rtc_alarm_irq_callback callback;
+	void *user_data;
+
+	struct k_mutex lock;
+	struct k_sem cr_upd_ack_sem;
+};
+
+static void rtc_atmel_sam_disable_wp(void)
+{
+	RTC_ATMEL_SAM_REG_RTC_WPMR = RTC_ATMEL_SAM_WPMR_DISABLE;
+}
+
+static void rtc_atmel_sam_enable_wp(void)
+{
+	RTC_ATMEL_SAM_REG_RTC_WPMR = RTC_ATMEL_SAM_WPMR_ENABLE;
+}
```

Comment:
It does, i checked to make sure since the lacking documentation regarding this register made it seem like it might not work, but indeed, if wp is not enabled/disabled like this, the MR register can not be updated :)